aten/src/ATen/core/interned_strings.h

```diff
@@ -285,9 +286,9 @@ namespace c10 {
   _(aten, zero_)                     \
   _(aten, fill_)                     \
   _(aten, masked_fill_)              \
-  _(cuda, _set_device)               \
-  _(cuda, set_stream)                \
-  _(cuda, _current_device)           \
+  _(_cuda, set_device)               \
```

Comment:
I'm pretty confused that *everything* got renamed to `_cuda` here. I thought @SplitInfinity was talking about a name conflict as the reason you might do `_cuda` https://github.com/pytorch/pytorch/pull/50488#discussion_r563982052 but that hardly seems relevant if you're renaming *everything*. Is there something I'm missing here?

Reply:
(If there isn't any conflict, isn't "cuda" (1) less changes and (2) a better namespace name?)

Reply:
There are C++ functions with the C++ qualified name `c10::cuda::current_device` and `c10::cuda::set_device` that conflict with the qualified names for the corresponding JIT operators when they are put into the `cuda` namespace (because `c10::Symbols` are also put into `c10`). I see two options:

- we can rename the entire namespace to `_cuda`
- we can add some code to translate the names of specific operators (like `set_device`) to names that won't conflict (e.g. `_set_device)` and keep the name `cuda` for the the namespace of the JIT CUDA operators

@nikithamalgifb is that correct?

Reply:
Yes that's right. Thanks for the explaination

Reply:
OK, so flipping the question around: why do these symbols need to be interned :)

Reply:
If by "interned" you mean represented as C++ symbols named `c10::cuda::XXX`, we need to mark the operators they represent as having side effects so that they don't get DCE'd. The code that handles that is a big switch statement and so these symbols must be interned so that they can be used as cases in that switch statement.